src/json/token.cr

```diff
@@ -17,16 +33,70 @@ class JSON::Token
     @raw_value = ""
   end
 
+  @[Deprecated("Use JSON::Token#kind, which is an enum")]
+  def type : Symbol
+    case @kind
+    when .null?
+      :null
+    when .false?
+      :false
+    when .true?
+      :true
+    when .int?
+      :INT
+    when .float?
+      :FLOAT
+    when .string?
+      :STRING
+    when .begin_array?
+      :"["
+    when .end_array?
+      :"]"
+    when .begin_object?
+      :"{"
+    when .end_object?
+      :"}"
+    when .colon?
+      :":"
+    when .comma?
+      :","
+    when .eof?
+      :EOF
+    else
+      raise "Unknown token kind: #{@kind}"
+    end
+  end
+
   def to_s(io : IO) : Nil
-    case @type
-    when :INT
-      @int_value.to_s(io)
-    when :FLOAT
-      @float_value.to_s(io)
-    when :STRING
-      @string_value.to_s(io)
+    case @kind
+    when .null?
+      io << "null"
+    when .false?
+      io << "false"
+    when .true?
+      io << true
+    when .int?
+      raw_value.to_s(io)
+    when .float?
+      raw_value.to_s(io)
+    when .string?
+      string_value.to_s(io)
+    when .begin_array?
+      io << "["
```

Comment:
All these could be `Char`s.

Reply:
Yes, but appending a Char is slower! The char needs to be decoded from its codepoint into UTF-8. With a string it's just a buffer copy.

Reply:
Actually, it seems appending a char is faster for some reason... maybe for a hardcoded char LLVM optimizes it better.